File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nombres: List[int]) -> List[int]:
        taille = len(nombres)
        resultat = [1] * taille

        facteur = 1
        for i in range(taille):
            resultat[i] = facteur
            facteur *= nombres[i]

        facteur = 1
        for i in range(taille - 1, -1, -1):
            resultat[i] *= facteur
            facteur *= nombres[i]

        return resultat

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int temp=nums.length;
        int [] res = new int[temp];
        res[0]=1;
        for (int i = 1; i<temp;i++){
            res[i]=nums[i-1]*res[i-1];
        }
        int suffix =1;
        for (int i = temp-1; i>-1 ;i--){
            res[i]=res[i]*suffix;
            suffix*=nums[i];
        }
        return res;
    }
}
